packages/cip30/src/Wallet/Wallet.ts

```diff
@@ -24,67 +31,88 @@ export type RequestAccess = () => Promise<boolean>;
 
 export type WalletOptions = {
   logger?: Logger;
-  persistAllowList?: boolean;
-  storage?: Storage;
+  storage?: Storage.LocalStorageArea;
 };
 
 const defaultOptions = {
   logger: dummyLogger,
   persistAllowList: false,
-  storage: window.localStorage
+  storage: storage.local
+};
+
+type WalletStorage = {
+  allowList: string[];
 };
 
 export class Wallet {
-  readonly version: SpecificationVersion;
+  readonly apiVersion: ApiVersion;
   readonly name: WalletName;
+  readonly icon: WalletIcon;
 
   private allowList: string[];
   private logger: Logger;
   private readonly options: Required<WalletOptions>;
 
-  constructor(
+  private constructor(
     properties: WalletProperties,
     private api: WalletApi,
     private requestAccess: RequestAccess,
+    allowList: string[],
     options?: WalletOptions
   ) {
     this.options = { ...defaultOptions, ...options };
     this.name = properties.name;
-    this.version = properties.version;
-    this.allowList = this.options.persistAllowList ? this.getAllowList() : [];
+    this.apiVersion = properties.apiVersion;
+    this.icon = properties.icon;
+    this.allowList = allowList;
     this.logger = this.options.logger;
   }
 
-  public getPublicApi(window: WindowMaybeWithCardano) {
+  public getPublicApi(hostname: string) {
```

Comment:
If the private members are designated, this method is probably redundant, as we can just use the class instance as a whole.